public/pages/Dashboard/Components/AnomaliesLiveChart.tsx

```diff
@@ -137,132 +161,200 @@ export const AnomaliesLiveChart = (props: AnomaliesLiveChartProps) => {
+                  size="s"
+                  title={`You are viewing ${MAX_LIVE_DETECTORS} detectors with the most recent anomaly occurrences.`}
+                  style={{
+                    width: '88%', // ensure width reaches NOW annotation line
+                    marginTop: '20px',
+                    marginBottom: '20px',
+                  }}
+                >
+                  <p>
+                    {`${MAX_LIVE_DETECTORS} detectors with the most recent anomalies are shown on the
+                    chart. Adjust filters if there are specific detectors you
+                    would like to monitor.`}
+                  </p>
+                </EuiCallOut>
+              ) : anomalousDetectorCount === 0 ? (
+                // all the data points have anomaly grade as 0
+                <EuiCallOut
+                  color="success"
+                  size="s"
+                  title="No anomalies found during the last 30 minutes across all matching detectors."
+                  style={{
+                    width: '96%', // ensure width reaches NOW line
+                    marginTop: '20px',
+                    marginBottom: '20px',
+                  }}
+                />
+              ) : null,
+              <div
+                style={{
+                  height: '200px',
+                  width: '100%',
+                  opacity: 1,
+                }}
+              >
+                <Chart>
+                  <Settings
+                    // hide legend if there only exists anomalies with 0 anomaly grade
+                    showLegend={!isEmpty(liveAnomalyData)}
+                    legendPosition={Position.Right}
+                    xDomain={{
+                      min: liveTimeRange.startDateTime.valueOf(),
+                      max: liveTimeRange.endDateTime.valueOf(),
+                    }}
+                  />
+                  <LineAnnotation
+                    domainType={AnnotationDomainTypes.XDomain}
+                    dataValues={annotations}
+                    style={TIME_NOW_LINE_STYLE}
+                    marker={'Now'}
+                  />
+                  <Axis
+                    id={'bottom'}
+                    position={Position.Bottom}
+                    tickFormat={timeFormatter}
+                    showOverlappingTicks={false}
+                  />
+                  <Axis
+                    id={'left'}
+                    title={'Anomaly grade'}
+                    position={Position.Left}
+                    domain={{ min: 0, max: 1 }}
+                  />
+                  <BarSeries
+                    // `id` for placeholder data point introduced by `prepareVisualizedAnomalies` shows as legend,
+                    // When there exists anomalies with anomaly grade > 0
+                    // we make `id` to blank string to hide the legend of placeholder data point
+                    id={!isEmpty(liveAnomalyData) ? '' : ' '}
```

Comment:
just curious: only blank string can hide legend ? Should we set a normal id rather than whitespace when the `liveAnomalyData` not empty?

Reply:
This is very tricky. 

If detectorName value for placeholder data is set as `null`, the `id` of `BarSeries` will be shown as legend for placeholder data.

Given empty `liveAnomalyData`, the legend of placeholder data will be hidden by changing `showLegend` to false, aka `showLegend={!isEmpty(liveAnomalyData)}`.  Thus we don't need to worry about legend of placeholder data in case of empty `liveAnomalyData`.

Given non-empty `liveAnomalyData`, the legend of placeholder data is shown along with legend of `liveAnomalyData`. Legend of placeholder data is not needed now. Changing `id` of `BarSeries` to blank string can make legend of placeholder data disappear, and normal legend of `liveAnomalyData` is still there. 

But we cannot always pass blank string `{''}`to `id`, as in the case of `null` detectorName + `id` as `{''}` , EuiChart will show `No data to dsiplay`, which UX wants to avoid. That's why  I pass `{' '}` /`{''}` to `id`

Reply:
After second thought, I changed to 
```
[AD_DOC_FIELDS.DETECTOR_NAME]: !isEmpty(liveAnomalyData) ? '' : null,
```

and keep using
```
<BarSeries
id={'Detector Anomaly grade'}`
```
It can also work. will change to that implementation and post another commit.